src/decorator/decorators.ts

```diff
@@ -0,0 +1,1036 @@
+        const args: ValidationMetadataArgs = {
+            type: ValidationTypes.NOT_CONTAINS,
+            target: object.constructor,
+            propertyName: propertyName,
+            value1: seed,
+            validationOptions: validationOptions
+        };
+        getFromContainer(MetadataStorage).addValidationMetadata(new ValidationMetadata(args));
+    };
+}
+
+/**
+ * Checks if the string contains only letters (a-zA-Z).
+ */
+export function IsAlpha(validationOptions?: ValidationOptions) {
+    return function (object: Object, propertyName: string) {
+        const args: ValidationMetadataArgs = {
+            type: ValidationTypes.IS_ALPHA,
+            target: object.constructor,
+            propertyName: propertyName,
+            validationOptions: validationOptions
+        };
+        getFromContainer(MetadataStorage).addValidationMetadata(new ValidationMetadata(args));
+    };
+}
+
+/**
+ * Checks if the string contains only letters and numbers.
+ */
+export function IsAlphanumeric(validationOptions?: ValidationOptions) {
+    return function (object: Object, propertyName: string) {
+        const args: ValidationMetadataArgs = {
+            type: ValidationTypes.IS_ALPHANUMERIC,
+            target: object.constructor,
+            propertyName: propertyName,
+            validationOptions: validationOptions
+        };
+        getFromContainer(MetadataStorage).addValidationMetadata(new ValidationMetadata(args));
+    };
+}
+
+/**
+ * Checks if the string contains ASCII chars only.
+ */
+export function IsAscii(validationOptions?: ValidationOptions) {
+    return function (object: Object, propertyName: string) {
+        const args: ValidationMetadataArgs = {
+            type: ValidationTypes.IS_ASCII,
+            target: object.constructor,
+            propertyName: propertyName,
+            validationOptions: validationOptions
+        };
+        getFromContainer(MetadataStorage).addValidationMetadata(new ValidationMetadata(args));
+    };
+}
+
+/**
+ * Checks if a string is base64 encoded.
+ */
+export function IsBase64(validationOptions?: ValidationOptions) {
+    return function (object: Object, propertyName: string) {
+        const args: ValidationMetadataArgs = {
+            type: ValidationTypes.IS_BASE64,
+            target: object.constructor,
+            propertyName: propertyName,
+            validationOptions: validationOptions
+        };
+        getFromContainer(MetadataStorage).addValidationMetadata(new ValidationMetadata(args));
+    };
+}
+
+/**
+ * Checks if the string's length (in bytes) falls in a range.
+ */
+export function IsByteLength(min: number, max?: number, validationOptions?: ValidationOptions) {
+    return function (object: Object, propertyName: string) {
+        const args: ValidationMetadataArgs = {
+            type: ValidationTypes.IS_BYTE_LENGTH,
+            target: object.constructor,
+            propertyName: propertyName,
+            value1: min,
```

Comment:
A refactoring thought here - it would be a cleaner api to define the `value*` into an array, ie 
`constraints: [min, max]` - that way the interface for `ValidationMetadataArgs` can easily handle *n* constraints. Also with the `...` spread operator, the arguments can be passed to the actual validator function as regular args.